02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md

```diff
@@ -193,6 +193,16 @@ Here's what has been added to the above since then:
 
 ## New Learners: What is the Best Way to Learn PureScript?
 
+There are likely other learning paths besides the ones covered below, but here's what we found in our research:
+
+![Learner Paths "Learner Paths"](./learner-paths-to-ps.png)
+
+| If your language background is... | ... then try these learning resources... | ... and these PureScript libraries |
+| -- | -- | -- |
+| JavaScript / OO languages | [Make the Leap from JavaScript to PureScript (Tutorial Series)](https://medium.com/@kelleyalex/index-make-the-leap-from-javascript-to-purescript-a1566d657e9c)<br><br>[An Outsider's Guide to Statically Typed Functional Programming (Book)](https://leanpub.com/outsidefp)<br><br>[Haskell: From First Principles](http://haskellbook.com/) | --
```

Comment:
Might be worth pointing at FantasyLand for PS learners coming from JS.
https://github.com/fantasyland/fantasy-land

Lots of PS type classes is abstract algebra stuff, which is what FantasyLand has implemented in a JS syntax, and with plenty of docs, IIRC.

Reply:
Hmm... first I've seen of it. I think it's better to include this than not.

Reply:
On second thought, I'm going to have to argue against adding this (or at least only adding this with a few "WARNINGS" added). 

For example, 
- A few of their explanations on FP Syntax seems misleading. For example, in [Type Signature Notation](https://github.com/fantasyland/fantasy-land#type-signature-notation), they explain `~>` as a `method type constructor`. I'm not sure that's really the best way to explain that... Moreover, my learning repo's overview of PS syntax is clearer than their examples/explanations (IMO).
- Rather than using `Eq`, they use `Setoid`
- Rather than using `Bind`, they use `Chain`/`ChainRec`
- They include a `Filterable` type class. PS does not have such a thing AFAIK

There's also [Static Land](https://github.com/rpominov/static-land), which differs somewhat. However, I think the JS language as a whole is part of the issue for me. That might be baggage I bring when reading through these resources

Reply:
Yeah, those are good points. Perhaps we could link to specific explanations of PureScript concepts in the FantasyLand project, rather than recommending the project as a whole? We could place this policy as the warning item next to it, and also add it to a list of resources to pull from/reference when people need to PureScript docs in the future.

Reply:
I think I'm still against this in general. 

If the issue is that JS-background people don't have a good side-by-side comparison for them that compares JS and PS, perhaps we just need to add one? (For example, this demonstrates the `do notation` for [Maybe, Either, and List](https://github.com/JordanMartinez/purescript-jordans-reference/blob/latestRelease/21-Hello-World/02-Prelude-ish/02-Prelude/07-Control-Flow--Functor-to-Monad/12-Useful-Monads.md) Or, perhaps we just ask them to read through the resulting JS output of a compiled PS program? After all, it is just Javascript.

Building on a slightly-wrong-but-mostly-true foundation tends not to go well. It makes it harder to figure out which part is the wrong part and how much of one's understanding is dependent on that.

Reply:
I agree with that. I'll have to spend more time looking at this and alternatives more closely to be a better advocate for it, but yeah, it's better to err on the side of caution for now, especially because concepts in PureScript have a more strict & detailed definition than concepts in other languages, so it's important to avoid sharing misleading teaching materials.